### PR TITLE
Parent test now an abstract parent test

### PR DESCRIPTION
**Pull request by srijan-sivakumar**
The Parent test has been converted into
an abstract class with certain methods
being abstract.

Now, the following changes have been incorporated
in this patch:
1. Exception handling now part of parent test class
and hence the test cases need not concern themselves
of the try except logic.
2. The test cases have been modified to reflect this latest
change - wrt Pt. 1.
3. A test case removed.
4. Traceback has been added into the parent test functionality.
This will be added to log whenever an exception fails a TC.

Updates: #179
Signed-off-by: srijan-sivakumar <ssivakum@redhat.com>

### CODE REVIEW

**Review by aujjwal-redhat — APPROVED** [review]
LGTM.

File: tests/functional/glusterd/test_peer_probe_detach.py [hunk 1]
```diff
@@ -12,28 +13,14 @@ class TestCase(ParentTest):
...
-            server_list = [server1,server2]
-            self.redant.create_cluster(server_list)
-
-        except Exception as e:
-            self.TEST_RES = False
-            print(f"Test is failed:{e}")
+        redant.create_cluster(self.server_list)
+        redant.pool_list(server1)
+        redant.peer_detach(server1, server2)
+        redant.peer_detach(server2, server2)
```
Comment (nik-redhat) [inline tests/functional/glusterd/test_peer_probe_detach.py:26]: The build error for these line seems genuine :thinking: @srijan-sivakumar

Comment (srijan-sivakumar) [reply tests/functional/glusterd/test_peer_probe_detach.py:26]: Aaah..Yes..The details about server1 and server2 have to be taken up from the server_list. Will send out a commit.

**Review by nik-redhat — APPROVED** [review]
lgtm

**Pull request merged**